src/main/java/fi/otavanopisto/kuntaapi/server/controllers/PageController.java

```diff
@@ -0,0 +1,137 @@
+import fi.otavanopisto.kuntaapi.server.id.OrganizationId;
+import fi.otavanopisto.kuntaapi.server.id.PageId;
+import fi.otavanopisto.kuntaapi.server.index.PageSearcher;
+import fi.otavanopisto.kuntaapi.server.index.SearchResult;
+import fi.otavanopisto.kuntaapi.server.integrations.AttachmentData;
+import fi.otavanopisto.kuntaapi.server.integrations.PageProvider;
+import fi.otavanopisto.kuntaapi.server.rest.model.Attachment;
+import fi.otavanopisto.kuntaapi.server.rest.model.LocalizedValue;
+import fi.otavanopisto.kuntaapi.server.rest.model.Page;
+
+@Dependent
+public class PageController {
+  
+  @Inject
+  private Logger logger;
+  
+  @Inject
+  private PageSearcher pageSearcher;
+
+  @Inject
+  private Instance<PageProvider> pageProviders;
+
+  public List<Page> listPages(OrganizationId organizationId, String path, boolean onlyRootPages, PageId parentId, Long firstResult, Long maxResults) {
+    List<Page> result = new ArrayList<>();
+    
+    for (PageProvider pageProvider : getPageProviders()) {
+      List<Page> pages = pageProvider.listOrganizationPages(organizationId, parentId, onlyRootPages, path);
+      if (pages != null) {
+        result.addAll(pages);
+      } else {
+        logger.severe(String.format("Page provider %s returned null when listing pages", pageProvider.getClass().getName())); 
+      }
+    }
+    
+    int resultCount = result.size();
+    int firstIndex = firstResult == null ? 0 : Math.min(firstResult.intValue(), resultCount);
+    int toIndex = maxResults == null ? resultCount : Math.min(firstIndex + maxResults.intValue(), resultCount);
+    
+    return result.subList(firstIndex, toIndex);
+  }
+
+  public List<Page> searchPages(OrganizationId organizationId, String queryString, Long firstResult, Long maxResults) {
+    SearchResult<PageId> searchResult = pageSearcher.searchPages(organizationId.getId(), queryString, firstResult, maxResults);
+    if (searchResult != null) {
+      List<Page> result = new ArrayList<>(searchResult.getResult().size());
+      
+      for (PageId pageId : searchResult.getResult()) {
+        Page page = findPage(organizationId, pageId);
+        if (page != null) {
+          result.add(page);
+        }
+      }
+      
+      return result;
+    }
+    
+    return Collections.emptyList();
+  }
+  
+  public Page findPage(OrganizationId organizationId, PageId pageId) {
+    for (PageProvider pageProvider : getPageProviders()) {
+      Page page = pageProvider.findOrganizationPage(organizationId, pageId);
+      if (page != null) {
+        return page;
+      }
+    }
+    
+    return null;
+  }
+
+  public List<LocalizedValue> getPageContents(OrganizationId organizationId, PageId pageId) {
+    for (PageProvider pageProvider : getPageProviders()) {
+      List<LocalizedValue> pageContents = pageProvider.findOrganizationPageContents(organizationId, pageId);
+      if (pageContents != null) {
+        return pageContents;
+      }
+    }
+    
+    return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)